### PR TITLE
Certora rules and migration fixes

### PR DESCRIPTION
**Pull request by 3esmit**
## Description

- Implement migration properly
- Fix calculation on rewards (now it reduces epoch supply and epoch reward), to keep all accounts getting a fair share
- Fix bug that reverted the processAccount due math overflow, when user process lower epochs than currentEpoch
- Fix bug that made pastEpochs to not update totalSupply when processing late accounts.

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Added natspec comments?
- [x] Ran `forge snapshot`?
- [x] Ran `pnpm lint`?
- [x] Ran `forge test`?
- [x] Ran `pnpm verify`?

### CODE REVIEW

File: contracts/StakeManager.sol [hunk 1]
```diff
@@ -57,6 +57,27 @@ contract StakeManager is Ownable {
...
+        _;
+    }
+
+    modifier onlyOldManager() {
+        if (msg.sender != address(oldManager)) {
+            revert StakeManager__SenderIsNotPreviousStakeManager();
+        }
+        _;
+    }
+
```
Comment (3esmit) [inline contracts/StakeManager.sol:80]: Implement migration properly

File: contracts/StakeManager.sol [hunk 2]
```diff
@@ -174,13 +223,24 @@ contract StakeManager is Ownable {
...
-        accounts[_vault] = _account;
+    }
+    /**
+     * @dev Only callable from old manager.
+     * @notice Increase total MP from old manager
+     * @param _increasedMP amount MP increased on account after migration initialized
+     */
+
+    function increaseMPFromMigration(uint256 _increasedMP) external onlyOldManager {
+        multiplierSupply += _increasedMP;
```
Comment (3esmit) [inline contracts/StakeManager.sol:243]: Implement migration properly

**Comment by 0x-r4bbit** [comment]
Friendly ping that this needs to be reviewed (I've cleaned up the commits).
There is one question re: `simplification()` I still need feedback on (see: https://discord.com/channels/864066763682218004/1143915396462809108/1204750696918814792)

Can update the PR once clarified

**Comment by 0x-r4bbit** [comment]
Closing this as this has already landed via #59

**Pull request closed without merging**